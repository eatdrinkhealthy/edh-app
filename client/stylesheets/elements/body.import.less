body {
   font-family: "Roboto", sans-serif;
   background-color: white;
   height: 100%;
   width: 100%;
   overflow-x: hidden;
   &.unselectable {
      -webkit-touch-callout: none;
      -webkit-user-select: none;
      user-select: none;
      -webkit-tap-highlight-color: rgba(0,0,0,0);
      -webkit-tap-highlight-color: transparent;
   }
}
