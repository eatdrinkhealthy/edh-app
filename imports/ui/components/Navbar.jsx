// @flow
import React from "react";
import { Link } from "react-router-dom";

const Navbar = (): React$Element<*> => (
  <div className="nav">
    <div className="dt w-100 pa2">
      <div className="dtc v-mid white w-25">Eat Drink Healthy</div>
      <div className="dtc v-mid w-75 tr">
        <Link
          id="navbarJoinLink"
          className="f6 link dim ma1 ph3 pv2 mb2 dib white bg-purple"
          to="/sidebar"
        >
          Join
        </Link>
        <Link
          id="navbarFilterLink"
          className="f6 link dim ma1 ph3 pv2 mb2 dib white bg-dark-green"
          to="/filter"
        >
          Filter
        </Link>
      </div>
    </div>
  </div>
);

export default Navbar;
