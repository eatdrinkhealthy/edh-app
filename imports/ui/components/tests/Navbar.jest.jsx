// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import { MemoryRouter } from "react-router-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Navbar from "../Navbar";
import { elements as es } from "../../../../tests/end-to-end/elements";

describe("<Navbar />", function() {
  it("matches render snapshot - with username", function() {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navbar username="testuser" userLoggedIn logout={() => {}} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches render snapshot - with null username", function() {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navbar username="" logout={() => {}} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("when user logged in should show username", function() {
    const wrapper = shallow(
      <Navbar userLoggedIn username="testuser" logout={jest.fn()} />,
    );
    expect(wrapper.find(es.navbar.username).prop("children")).toEqual(
      "Welcome, testuser!",
    );
  });

  it("when user not logged in should not show username", function() {
    const wrapper = shallow(<Navbar username="" logout={jest.fn()} />);
    expect(wrapper.find(es.navbar.username).prop("children")).toEqual(""); // no username
  });
});
