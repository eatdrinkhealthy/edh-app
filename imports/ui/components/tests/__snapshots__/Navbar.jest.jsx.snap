// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> matches render snapshot - with null username 1`] = `
<div
  className="nav__bar"
  id="navbar"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4 col-xs-6"
      >
        <img
          alt="Eat Drink Healthy logo"
          className="nav__logo"
          src="/images/EDH-logo.png"
        />
        <span
          className="nav__title"
        >
          EAT. DRINK. HEALTHY.
        </span>
      </div>
      <div
        className="nav__username col-md-5 hidden-sm hidden-xs"
        id="loggedInUser"
      >
        
      </div>
      <div
        className="col-md-3 col-xs-6"
      >
        <span
          className="nav__usermenu"
        >
          <span>
            <a
              className="btn join_button"
              href="/sidebar?action=signup"
              id="joinLink"
              onClick={[Function]}
            >
              JOIN
            </a>
            <a
              className="btn log_in_out_button"
              href="/sidebar?action=login"
              id="loginLink"
              onClick={[Function]}
            >
              LOG IN
            </a>
          </span>
        </span>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="nav__username_row2 hidden-lg hidden-md"
      >
        
      </div>
    </div>
  </div>
</div>
`;

exports[`<Navbar /> matches render snapshot - with username 1`] = `
<div
  className="nav__bar"
  id="navbar"
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-md-4 col-xs-6"
      >
        <img
          alt="Eat Drink Healthy logo"
          className="nav__logo"
          src="/images/EDH-logo.png"
        />
        <span
          className="nav__title"
        >
          EAT. DRINK. HEALTHY.
        </span>
      </div>
      <div
        className="nav__username col-md-5 hidden-sm hidden-xs"
        id="loggedInUser"
      >
        Welcome, testuser!
      </div>
      <div
        className="col-md-3 col-xs-6"
      >
        <span
          className="nav__usermenu"
        >
          <a
            className="btn log_in_out_button"
            href=""
            id="logoutLink"
            onClick={[Function]}
          >
            LOG OUT
          </a>
        </span>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="nav__username_row2 hidden-lg hidden-md"
      >
        Welcome, testuser!
      </div>
    </div>
  </div>
</div>
`;
